scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -301,252 +321,266 @@
-                                            ; That's good for (almost) all parts of the system. However,
-                                            ; If it is legitimately missing, then something weird is going on. If it is
-                                            ; deleting, that's an expected state. So, let's be selective with our logging.
-                                            (if (= (:state synthesized-state) :missing)
-                                              (log/info "In compute cluster" name ", pod" pod-name
-                                                        "was killed with cook expected state"
-                                                        (prepare-cook-expected-state-dict-for-logging cook-expected-state-dict)
-                                                        "and k8s actual state"
-                                                        (prepare-k8s-actual-state-dict-for-logging k8s-actual-state-dict))
-                                              (log-weird-state compute-cluster pod-name
-                                                               cook-expected-state-dict k8s-actual-state-dict))
-                                            (handle-pod-killed compute-cluster pod-name)))
-                                        :pod/failed (handle-pod-completed compute-cluster pod-name k8s-actual-state-dict)
-                                        :pod/running (kill-pod api-client cook-expected-state-dict pod)
-                                        ; There was a race and it completed normally before being it was killed.
-                                        :pod/succeeded (handle-pod-completed compute-cluster pod-name k8s-actual-state-dict)
-                                        ; TODO: Should mark mea culpa retry
-                                        :pod/unknown (handle-pod-completed-and-kill-pod-in-weird-state
-                                                       compute-cluster pod-name k8s-actual-state-dict)
-                                        :pod/waiting (kill-pod api-client cook-expected-state-dict pod))
-
-                                      :cook-expected-state/running
-                                      (case pod-synthesized-state-modified
-                                        ; This indicates that something deleted it behind our back
-                                        :missing (do
-                                                   (log/info "In compute cluster" name ", something deleted"
-                                                              pod-name "behind our back")
+  ; How long is it taking to process this? Note that the lock is held, so we can detect contention if this
+  ; is larger than process-lock, which includes the locking time too.
+  ; If the product of the rate * time exceeds 1, we have contention. We should lock shard if it exceeds .3.
```

Comment:
In my opinion, we should remove the comment. We don't need to justify our metrics. Suggestions for analysis also don't seem appropriate here.

Reply:
Nuked.